client/components/experiment/index.tsx

```diff
@@ -22,7 +23,21 @@ export { default as LoadingVariations } from './loading-variations';
  * @param props The properties that describe the experiment
  */
 export const Experiment: FunctionComponent< ExperimentProps > = ( props ) => {
-	const { isLoading: loading, variation, children } = props;
+	const { isLoading: loading, variation, children, name: experimentName } = props;
+	const [ eventFired, setEventFired ] = useState< boolean >( false );
+	useEffect( () => {
+		setEventFired( true );
+	}, [] );
+
+	if ( ! eventFired ) {
+		// Do to how tracks works, we need to always fire an event immediately to generate an anonid. This event is here
+		// to guarantee that we have an anonid if the browser needs one.
+
+		recordTracksEvent( 'calypso_experiment_rendered', {
+			experimentName,
```

Comment:
Should this be a key/value pair? something like `experiment_name: experimentName,`.

Reply:
Probably &ndash; Tracks prop names have to be in snake case.

Reply:
Oh yeah!